src/fides/api/service/connectors/mongodb_connector.py

```diff
@@ -27,23 +27,14 @@ def build_uri(self) -> str:
         """
 
         config = MongoDBSchema(**self.configuration.secrets or {})
-
-        user_pass: str = ""
-        default_auth_db: str = ""
-        if config.username and config.password:
-            user_pass = f"{config.username}:{config.password}@"
-
-            if config.defaultauthdb:
-                default_auth_db = f"/{config.defaultauthdb}"
-
-        port: str = f":{config.port}" if config.port else ""
-        url = f"mongodb://{user_pass}{config.host}{port}{default_auth_db}"
+        user_pass = f"{config.username}:{config.password}@"
+        default_auth_db = f"/{config.defaultauthdb}"
+        url = f"mongodb://{user_pass}{config.host}{config.port}{default_auth_db}"
         return url
 
     def create_client(self) -> MongoClient:
         """Returns a client for a MongoDB instance"""
-        config = MongoDBSchema(**self.configuration.secrets or {})
-        uri = config.url if config.url else self.build_uri()
+        uri = self.build_uri()
```

Comment:
`url` is no longer available, so we must always call `build_uri` to build the URL from the provided components.